xonsh/completers/path.py

```diff
@@ -151,6 +151,59 @@ def _quote_paths(paths, start, end):
     return out
 
 
+def joinpath(path):
+    if path is None:
+        return ''
+    if len(path) == 0:
+        return ''
+    if path == ('',):
+        return get_sep()
+    elif path[0] == '':
+        return get_sep() + _normpath(os.path.join(*path))
+    else:
+        return _normpath(os.path.join(*path))
+
+
+def splitpath(path):
+    path = _normpath(path)
+    if path.startswith(get_sep()):
+        pre = ('', )
+    else:
+        pre = ()
+    return pre + _splitpath(path, ())
+
+
+def _splitpath(path, sofar=()):
+    folder, path = os.path.split(path)
+    if path == "":
+        return sofar[::-1]
+    elif folder == "":
+        return (sofar + (path, ))[::-1]
+    else:
+        return _splitpath(folder, sofar + (path, ))
+
+
+def expanded_match(ref, typed):
```

Comment:
> what if you switch letters ?

I had something like that in the first commit, but I couldn't quite get it to work as I expected and it led to a lot of matches that I think wouldn't make sense to a lot of people.  Maybe that's because I was using Levenshtein distance, and maybe `difflib` would give some better results.